File: m3-h5/src/main/java/ru/yegorpilipenko/otus/spring/m2h4/controller/AuthorController.java
package ru.yegorpilipenko.otus.spring.m2h4.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import ru.yegorpilipenko.otus.spring.m2h4.model.Author;
import ru.yegorpilipenko.otus.spring.m2h4.service.AuthorService;

import java.util.Collection;

import static ru.yegorpilipenko.otus.spring.m2h4.controller.AuthorController.URL;

@RestController
@RequestMapping(URL)
@RequiredArgsConstructor
public class AuthorController {

    static final String URL = "/authors";

    private final AuthorService authorService;

    @GetMapping
    public Collection<Author> getAll() {
        return authorService.getAll();
    }

}

File: m2-h3/src/main/kotlin/ru/yegorpilipenko/otus/spring/m2h3/controller/AuthorController.kt
package ru.yegorpilipenko.otus.spring.m2h3.controller

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import ru.yegorpilipenko.otus.spring.m2h3.service.AuthorService

@Controller
class AuthorController(val authorService: AuthorService) {

    @GetMapping("/authors")
    fun authors(model: Model): String {
        val authors = authorService.getAll()
        model.addAttribute("authors", authors)
        return "authors"
    }

}
